src/routing_membrane.rs

```diff
@@ -557,7 +561,12 @@ impl<F> RoutingMembrane<F> where F: Interface {
                         message.clone(), request.clone()),
                     MessageType::GetDataResponse(ref response) =>
                         self.handle_get_data_response(message_wrap, message.clone(), response.clone()),
-                    MessageType::PutData(ref data) => self.handle_put_data(message_wrap, message.clone(), data.clone()),
+                    MessageType::PutData(ref data) => {
+                        match message.source.actual_source() {
+                            Address::Node(name) => self.handle_group_put_data(message_wrap, message.clone(), data.clone(), name),
+                            Address::Client(name) => self.handle_client_put_data(message_wrap, message.clone(), data.clone(), name),
+                        }
+                    },
```

Comment:
> a node which is not fully joined the network, depending on the size of its routing table expects a minimum number of messages.

Sorry, I don't understand this argument. What I meant was when a normal node receives a message from a not-yet-connected node.

Though the real problem I see here is that we're assuming something from the upper layers (that a node never does non-group put request), if such assumption was valid, we should explicitly check it somewhere to make it clear that we require it. In here, the question is "does it come from a group or not" and `message.from_authority().is_group()` answers this question without any implicit assumptions (which may turn out true or false later).